driven/storage/adapter.go

```diff
@@ -124,16 +140,26 @@ func (sa Adapter) FindUsersByIDs(usersIDs []string) ([]model.User, error) {
 
 // FindUserByToken finds firebase token
 func (sa Adapter) FindUserByToken(orgID string, appID string, token string) (*model.User, error) {
-	return sa.findUserByTokenWithContext(context.Background(), orgID, appID, token)
+	return sa.findUserByTokenWithContext(context.Background(), orgID, appID, token, nil)
 }
 
-func (sa Adapter) findUserByTokenWithContext(context context.Context, orgID string, appID string, token string) (*model.User, error) {
+func (sa Adapter) findUserByTokenWithContext(context context.Context, orgID string, appID string, token string, tokenType *string) (*model.User, error) {
 	filter := bson.D{}
 	if len(token) > 0 {
-		filter = bson.D{
-			primitive.E{Key: "org_id", Value: orgID},
-			primitive.E{Key: "app_id", Value: appID},
-			primitive.E{Key: "firebase_tokens", Value: bson.D{primitive.E{Key: "$elemMatch", Value: bson.D{primitive.E{Key: "token", Value: token}}}}},
+		if tokenType != nil {
+			if *tokenType == "airship" {
```

Comment:
The token type passed into the storage function should just be used to prepend `"_tokens"` for use in the filter (for now it will be `"firebase"` or `"airship"` and not nullable). Storage functions should not do an explicit check like this on the token type because it should not depend on which services we choose to support tokens for.